Alicerce/Sources/Logging/Log+NodeLogDestination.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  Log+NodeLogDestination.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 07/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public extension Log {
+
+    public class NodeLogDestination : LogDestination {
+
+        public struct Errors: Error {
+            enum ErrorKind {
+                case httpError
+            }
+
+            let kind: ErrorKind
+            let httpStatus: Int
+        }
+
+        private static let dispatchQueueLabel = "com.mindera.Alicerce.NodeLogDestination"
+        private static let defaultRequestTimeout: TimeInterval = 0
+
+        public private(set) var dispatchQueue: DispatchQueue
+        public private(set) var minLevel: Level
+        public private(set) var formatter: LogItemFormatter
+
+        private let serverURL: URL
+        private let urlSession: URLSession
+        private let requestTimeout: TimeInterval
+
+        //MARK:- lifecycle
+
+        public init(serverURL: URL,
+                    minLevel: Level = Log.Level.error,
+                    formatter: LogItemFormatter = Log.StringLogItemFormatter(),
+                    urlSession: URLSession = URLSession.shared,
+                    dispatchQueue: DispatchQueue = DispatchQueue(label: NodeLogDestination.dispatchQueueLabel),
+                    requestTimeout: TimeInterval = NodeLogDestination.defaultRequestTimeout) {
+
+            self.serverURL = serverURL
+            self.minLevel = minLevel
+            self.formatter = formatter
+            self.urlSession = urlSession
+            self.dispatchQueue = dispatchQueue
+            self.requestTimeout = requestTimeout
+        }
+
+        //MARK:- public methods
+
+        public func write(item: Item, completion: @escaping (LogDestination, Log.Item, Error?) -> Void) {
+            dispatchQueue.async { [weak self] in
+                guard let strongSelf = self else { return }
+
+                var reportedError: Error?
```

Comment:
I might be missing something again, but I don't see anywhere where `reportedError` is changed before the scope returns?